### PR TITLE
Feature/dashboards panel

### LINKED ISSUE #1340: Dashboard - JCEF Plugin Integration
Dashboard document should be singleton per environment - should not open more than one per environment.

image

https://www.figma.com/file/9WRmS8uKjUIv45ldWun2pt/UI---Shared?type=design&node-id=2140-161640&mode=design&t=rRex5MywB4vCoK31-0

### PR DESCRIPTION
**Pull request by borisermakof**
Fixes #1340 

image

### CODE REVIEW

File: analytics-provider/src/main/java/org/digma/intellij/plugin/analytics/AnalyticsProvider.java [hunk 1]
```diff
@@ -89,4 +90,6 @@ public interface AnalyticsProvider extends Closeable {
     PerformanceMetricsResponse getPerformanceMetrics();
 
     DeleteEnvironmentResponse deleteEnvironment(DeleteEnvironmentRequest deleteEnvironmentRequest);
+
+    String getDashboard(Map<String,String> fields);
```
Comment (arik-dig) [inline analytics-provider/src/main/java/org/digma/intellij/plugin/analytics/AnalyticsProvider.java:94]: what is the siganature, can you doc what is the key and what is the value?

**Review by arik-dig — CHANGES_REQUESTED** [review]
please add PR outcome (UI print screen)

File: analytics-provider/src/main/java/org/digma/intellij/plugin/analytics/AnalyticsProvider.java (hunk 1 above)
Comment (borisermakof) [reply analytics-provider/src/main/java/org/digma/intellij/plugin/analytics/AnalyticsProvider.java]: added

File: analytics-provider/src/main/java/org/digma/intellij/plugin/analytics/AnalyticsProvider.java [hunk 2]
```diff
@@ -89,4 +90,7 @@ public interface AnalyticsProvider extends Closeable {
     PerformanceMetricsResponse getPerformanceMetrics();
 
     DeleteEnvironmentResponse deleteEnvironment(DeleteEnvironmentRequest deleteEnvironmentRequest);
+
+    // queryParams: (limit, 5), (pageSize, 5) etc
+    String getDashboard(Map<String,String> queryParams);
```
Comment (arik-dig) [inline analytics-provider/src/main/java/org/digma/intellij/plugin/analytics/AnalyticsProvider.java:95]: is there a reason the signature is not strict one, with field names?

**Review by arik-dig — APPROVED** [review]

File: analytics-provider/src/main/java/org/digma/intellij/plugin/analytics/AnalyticsProvider.java (hunk 2 above)
Comment (borisermakof) [reply analytics-provider/src/main/java/org/digma/intellij/plugin/analytics/AnalyticsProvider.java:95]: its generic, depends on dashboard

File: src/main/java/org/digma/intellij/plugin/dashboard/DashboardMessageRouterHandler.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+                        backendQueryParams.put("type", dashboardType);
+                        // query parameters
+                        Map<String, Object> payloadQueryParams = (Map<String, Object>) payload.get("query");
+                        payloadQueryParams.forEach((paramKey, paramValue) -> {
+                            backendQueryParams.put(paramKey, paramValue.toString());
+                        });
+
+                        try {
+                            var dashboardJson = DashboardService.getInstance(project).getDashboard(backendQueryParams);
+                            Log.log(logger::trace, project, "got notifications {}", dashboardJson);
```
Comment (kshmidt-digma) [inline src/main/java/org/digma/intellij/plugin/dashboard/DashboardMessageRouterHandler.java:77]:
```suggestion
                            Log.log(logger::trace, project, "got dashboard {}", dashboardJson);
```

File: src/main/kotlin/org/digma/intellij/plugin/ui/dashboard/DashboardButton.kt [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+    private fun doActionListener() {
+
+        try {
+
+            DashboardService.getInstance(project).openDashboard("Dashboard Panel - " + getInstance().state.currentEnv);
+
+        } catch (e: Exception) {
+            Log.warnWithException(logger, project, e, "Error in doActionListener")
+            ErrorReporter.getInstance().reportError(project, "NotificationsButton.doActionListener", e)
```
Comment (kshmidt-digma) [inline src/main/kotlin/org/digma/intellij/plugin/ui/dashboard/DashboardButton.kt:52]:
```suggestion
            ErrorReporter.getInstance().reportError(project, "DasboardButton.doActionListener", e)
```

**Pull request merged**